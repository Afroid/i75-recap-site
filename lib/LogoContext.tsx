"use client";
import { createContext, useContext, useMemo } from "react";

const logos = [
  "/default-logo.png",
  "/default-logo-alt.png",
  "/default-RWB.png",
];

const LogoContext = createContext<string | null>(null);

export function LogoProvider({ children }: { children: React.ReactNode }) {
  // Randomizes only once on initial render
  const logo = useMemo(
    () => logos[Math.floor(Math.random() * logos.length)],
    []
  );
  return (
    <LogoContext.Provider value={logo}>
      {children}
    </LogoContext.Provider>
  );
}

export function useLogo() {
  const logo = useContext(LogoContext);
  if (logo === null) throw new Error("useLogo must be inside LogoProvider");
  return logo;
}
