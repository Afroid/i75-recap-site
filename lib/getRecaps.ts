import recap2024 from "@/data/recaps/2024.json";
import recap2023 from "@/data/recaps/2023.json";
import { RecapData } from "@/types/types";

export function getRecaps(): Record<number, number[]> {
  const allData: RecapData[] = [recap2024, recap2023]; // Add more as needed
  const summary: Record<number, number[]> = {};

  for (const season of allData) {
    summary[season.year] = season.recaps.map((r) => r.week);
  }

  return summary;
}
