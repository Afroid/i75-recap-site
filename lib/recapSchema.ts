import { z } from "zod";

const RecapMatchupSchema = z.object({
  team1: z.string(),
  team2: z.string(),
  winner: z.string(),
  score: z.string(),
  breakdown: z.string(),
  imageUrl: z.string().optional(),
  imageAlt: z.string().optional(),
});

const GameNotesSectionSchema = z.object({
  type: z.literal("gameNotes"),
  pointsLeader: z.object({
    name: z.string(),
    points: z.number(),
    notes: z.string().optional(),
  }),
  dud: z.object({
    name: z.string(),
    points: z.number(),
    notes: z.string().optional(),
  }),
  matchups: z.array(RecapMatchupSchema),
});

const RecapSectionSchema = z.discriminatedUnion("type", [
  z.object({
    type: z.literal("intro"),
    content: z.string(),
  }),
  z.object({
    type: z.literal("tidbits"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("injuries"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("sidePieces"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("matchups"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("farewell"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  })
]);

const RecapWeekSchema = z.object({
  week: z.number(),
  title: z.string(),
  sections: z.array(z.union([RecapSectionSchema, GameNotesSectionSchema])),
});

export const RecapDataSchema = z.object({
  year: z.number(),
  recaps: z.array(RecapWeekSchema),
});
