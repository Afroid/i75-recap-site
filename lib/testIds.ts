export const TestIds = {
  /**
   * Pages
   */

  // Index - App Level
  SITE_HEADER: "site-header",
  MAIN_CONTENT: "main-content",
  PAGE_WRAPPER: "page-wrapper",

  // Home Page
  HOME_MAIN:   "home-main",
  HOME_HERO_LOGO: "home-hero-logo",
  HOME_INTRO_SECTION: "home-intro-section",
  HOME_HERO_HEADING: "home-hero-heading",
  HOME_CTA_CONTAINER: "home-cta-container",
  HOME_VIEW_ALL_RECAPS_BUTTON: "home-view-all-recaps-button",
  HOME_LEARN_MORE_BUTTON: "home-learn-more-button",

  // 404 Page
  NOT_FOUND_MAIN: "not-found-main",
  RETURN_HOME_LINK: "return-home-link",

  // About Page
  ABOUT_MAIN: "about-main",

  // Contact Page
  CONTACT_MAIN: "contact-main",
  CONTACT_EMAIL_LINK: "contact-email-link",
  CONTACT_FORM: "contact-form",
  CONTACT_FORM_NAME_SECTION: "contact-form-name-section",
  CONTACT_FORM_EMAIL_SECTION: "contact-form-email-section",
  CONTACT_FORM_MESSAGE_SECTION: "contact-form-message-section",
  CONTACT_FORM_SUBMIT: "contact-form-submit",

  /**
   * Recap Pages - (pages/recaps)
   */

  // Recap Page

  // View All Recaps Page
  VIEW_ALL_RECAPS: "view-all-recaps",
  // Year Page

  /**
   * Components
   */

  // Breadcrumb Component
  BREADCRUMB: "breadcrumb",
  BREADCRUMB_NAV: "breadcrumb-nav",
  BREADCRUMB_LIST: "breadcrumb-list",
  BREADCRUMB_ITEM: "breadcrumb-item",
  BREADRCRUMB_LINK: "breadcrumb-link",
  BREADCRUMB_CURRENT: "breadcrumb-current",
  BREADCRUMB_SEPARATOR: "breadcrumb-separator",

  // DesktopNavbar Component

  // GameNotesSection.tsx
  GAME_NOTES_SECTION: "game-notes-section",
  GAME_NOTES_ITEM: "game-notes-item",
  GAME_NOTES_SCOREBOARD: "game-notes-scoreboard",
  GAME_NOTES_BREAKDOWN: "game-notes-breakdown",
  GAME_NOTES_IMAGE_CONTAINER: "game-notes-image-container",
  GAME_NOTES_IMAGE: "game-notes-image",

  // Header.tsx
  HEADER_SITE_HEADER: "site-header",
  HEADER_MOBILE_HEADER: "mobile-header",
  HEADER_HAMBURGER_BUTTON: "header-hamburger-button",
  HEADER_MOBILE_LOGO: "header-mobile-logo",
  HEADER_DESKTOP_LOGO: "header-desktop-logo",
  HEADER_DESKTOP_NAV: "header-desktop-nav",
  HEADER_MOBILE_DRAWER: "header-mobile-drawer",

  // HeaderLogo.tsx
  HEADER_LOGO_LINK: "header-logo-link",
  HEADER_LOGO_BG_DARK: "header-logo-bg-dark",
  HEADER_LOGO_BG_LIGHT: "header-logo-bg-light",
  HEADER_LOGO_CONTAINER: "header-logo-container",
  HEADER_LOGO_NAME: "header-logo-name",
  HEADER_LOGO_DYNAMIC: "header-logo-dynamic",

  // LeaderDudSection.tsx
  LDS_SECTION: "lds-section",
  LDS_LEADER_HEADING: "lds-leader-heading",
  LDS_LEADER_POINTS: "lds-leader-points",
  LDS_LEADER_NOTES: "lds-leader-notes",
  LDS_DUD_HEADING: "lds-dud-heading",
  LDS_DUD_POINTS: "lds-dud-points",
  LDS_DUD_NOTES: "lds-dud-notes",

  // MobileDrawer.tsx
  MOBILE_SEASON_TOGGLE: "mobile-season-toggle",
  MOBILE_SEASON_WEEKS: "mobile-season-weeks",
  MOBILE_WEEK_LINK: "mobile-week-link",
  MOBILE_DRAWER_OVERLAY: "mobile-drawer-overlay",
  MOBILE_DRAWER: "mobile-drawer",
  MOBILE_DRAWER_TITLE: "mobile-drawer-title",
  MOBILE_OLDER_SEASON_TITLE: "mobile-older-season-title",
  MOBILE_TOGGLE_OLDER_SEASONS: "mobile-toggle-older-seasons",

  // RecapLayout.tsx
  RECAP_TITLE: "recap-title",
  RECAP_SECTION: "recap-section",
  RECAP_HEADING: "recap-heading",
  RECAP_INTRO: "recap-intro",
  RECAP_BULLETS: "recap-bullets",
  RECAP_BULLET: "recap-bullet",
  RECAP_CONTENT: "recap-content",
  RECAP_IMAGE: "recap-image",

  // Scoreboard.tsx
  SCOREBOARD_CONTAINER: "scoreboard-container",
  SCOREBOARD_ENTRY: "scoreboard-entry",
  SCOREBOARD_LOGO: "scoreboard-logo",
  SCOREBOARD_NAME: "scoreboard-name",
  SCOREBOARD_RECORD: "scoreboard-record",
  SCOREBOARD_SCORE: "scoreboard-score",

  // TeamLogo.tsx
  TEAM_LOGO_IMAGE: "team-logo-image",

  // Static Links
  NAV_LINK_HOME: "nav-link-home",
  NAV_LINK_ABOUT: "nav-link-about",
  NAV_LINK_CONTACT: "nav-link-contact"
}
