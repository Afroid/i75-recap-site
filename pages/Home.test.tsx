import { render, screen } from '@testing-library/react'
import Home from '@/pages/index.page';
import { TestIds } from '@/lib/testIds';

describe('Home page', () => {
  it('asserts that the the hero heading renders', () => {
    render(<Home />)
    const heading = screen.getByRole('heading', { level: 1, name: /I75 League Recaps/i })
    expect(heading).toBeInTheDocument()
  })

  it('renders the hero heading text', () => {
    // Renders the page
    render(<Home />)

    // Grabs the element
    const heading = screen.getByTestId(TestIds.HOME_HERO_HEADING)

    // Assertions
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveTextContent(/I75 League Recaps/i)
  })

  it('matches the snapshot', () => {
    const { asFragment } = render(<Home />)
    expect(asFragment()).toMatchSnapshot()
  })
})
