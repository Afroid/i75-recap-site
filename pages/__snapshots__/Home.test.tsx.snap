// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches the snapshot 1`] = `
<DocumentFragment>
  <main
    data-testid="home-main"
  >
    <div
      class="mx-auto w-full max-w-lg"
      data-testid="home-hero-logo-container"
    >
      <img
        alt="Hero Logo"
        class="w-auto h-auto rounded drop-shadow-heavy"
        data-nimg="1"
        data-testid="home-hero-logo"
        decoding="async"
        height="400"
        src="/_next/image?url=%2Fdefault-logo.png&w=1200&q=75"
        srcset="/_next/image?url=%2Fdefault-logo.png&w=640&q=75 1x, /_next/image?url=%2Fdefault-logo.png&w=1200&q=75 2x"
        style="color: transparent;"
        width="600"
      />
    </div>
    <section
      class="max-w-2xl mx-auto text-center"
      data-testid="home-intro-section"
    >
      <h1
        class="lg:text-5xl text-3xl font-extrabold text-green-700 mb-4"
        data-testid="home-hero-heading"
      >
        I75 League Recaps
      </h1>
      <p
        class="text-lg text-gray-600 mb-6"
        data-testid="home-intro-text"
      >
        Welcome to the official recap hub for the I75 League, where every week’s matchups are transformed from email threads into a polished web experience—complete with scores, highlights, GIFs, and a dash of humor.
      </p>
      <div
        class="flex justify-center gap-4"
        data-testid="home-cta-container"
      >
        <a
          class="px-5 py-3 bg-green-600 text-white font-semibold rounded-lg hover:bg-green-700 transition"
          data-testid="home-view-all-recaps-button"
          href="/recaps/viewAllRecaps"
        >
          View All Recaps
        </a>
        <a
          class="px-5 py-3 border border-green-600 text-green-600 rounded-lg hover:bg-green-50 transition"
          data-testid="home-learn-more-button"
          href="/about"
        >
          Learn More
        </a>
      </div>
    </section>
  </main>
</DocumentFragment>
`;
